crates/exec-wasmtime/src/loader/configured/platform.rs

```diff
@@ -57,6 +59,7 @@ impl Platform {
 
         match (rax, rdx) {
             (ENOSYS, ..) => Ok(Self(Technology::Kvm, 0)),
```

Comment:
For the `KVM` shim or in this case the Linux kernel instead of one of our shims, the return value `ENOSYS` is expected, because it is our own invention for SGX and SNP shims, and the Linux kernel can't know about it. Therefore it is handled like in the `KVM` case and an `OK()` is returned. `EPERM` is probably a restriction `podman` installs via [seccomp filters](https://lwn.net/Articles/494252/).